File: init/fast/nightside/Jcurrent_gaussian_tanh.m
function E = Jcurrent_gaussian_tanh(E, Nt, gridflag, flagdip)
% Set the top boundary shape (current density) and potential solve type
% flag.  Can be adjusted by user to achieve any desired shape.

arguments
  E (1,1) struct
  Nt (1,1) {mustBePositive,mustBeInteger}
  gridflag (1,1) {mustBePositive,mustBeInteger} = 0 %#ok<INUSA>
  flagdip (1,1) logical = false %#ok<INUSA>
end

Jpk=E.Jtarg;
%mlonsig=E.Efield_lonwidth;
%mlatsig=E.Efield_latwidth;
displace=10*E.mlatsig;
mlatctr=E.mlatmean+displace*tanh((E.MLON-E.mlonmean)/(2*E.mlonsig));    %changed so the arc is wider compared to its twisting
for it=1:Nt
  E.flagdirich(it)=0;
  E.Vminx1it(:,:,it)=zeros(E.llon,E.llat);
  if (it>2)
    E.Vmaxx1it(:,:,it)=Jpk.*exp(-(E.MLON-E.mlonmean).^2/2/E.mlonsig^2).*exp(-(E.MLAT-mlatctr-1.5*E.mlatsig).^2/2/E.mlatsig^2);
    E.Vmaxx1it(:,:,it)=E.Vmaxx1it(:,:,it)-Jpk.*exp(-(E.MLON-E.mlonmean).^2/2/E.mlonsig^2).*exp(-(E.MLAT-mlatctr+1.5*E.mlatsig).^2/2/E.mlatsig^2);
  else
    E.Vmaxx1it(:,:,it)=zeros(E.llon,E.llat);
  end %if
end %for

end % function

File: init/arcs_python_AMR/Jcurrent_gaussian_tanh.py
import numpy as np
import xarray


def Jcurrent_gaussian_tanh(E: xarray.Dataset, gridflag: int, flagdip: bool) -> xarray.Dataset:
    """
    Set the top boundary shape (current density) and potential solve type
    flag.  Can be adjusted by user to achieve any desired shape.
    """

    Jpk = E.Jtarg
    llon = E.mlon.size
    llat = E.mlat.size

    displace = 10 * E.mlatsig
    mlatctr = E.mlatmean + displace * np.tanh((E.mlon - E.mlonmean) / (2 * E.mlonsig))
    # changed so the arc is wider compared to its twisting
    for i, t in enumerate(E.time):
        E["flagdirich"].loc[t] = 0
        E["Vminx1it"].loc[t] = np.zeros((llon, llat))
        if i > 2:
            E["Vmaxx1it"].loc[t] = (
                Jpk
                * np.exp(-((E.mlon - E.mlonmean) ** 2) / 2 / E.mlonsig**2)
                * np.exp(-((E.mlat - mlatctr - 1.5 * E.mlatsig) ** 2) / 2 / E.mlatsig**2)
            )
            E["Vmaxx1it"].loc[t] = E["Vmaxx1it"].loc[t] - Jpk * np.exp(
                -((E.mlon - E.mlonmean) ** 2) / 2 / E.mlonsig**2
            ) * np.exp(-((E.mlat - mlatctr + 1.5 * E.mlatsig) ** 2) / 2 / E.mlatsig**2)
        else:
            E["Vmaxx1it"].loc[t] = np.zeros((llon, llat))

    return E
